packages/tpetra/core/src/Tpetra_CrsGraph_decl.hpp

```diff
@@ -1195,6 +1206,19 @@ namespace Tpetra {
                     const Teuchos::ArrayView<const LocalOrdinal>& permuteToLIDs,
                     const Teuchos::ArrayView<const LocalOrdinal>& permuteFromLIDs) override;
 
+    void
+    resizeForIncomingDataImpl (const Kokkos::UnorderedMap<LocalOrdinal, size_t>& padding);
+
+    void
+    resizeForIncomingData (const RowGraph<LocalOrdinal, GlobalOrdinal, Node>& source,
```

Comment:
These two overloads of `resizeForIncomingData` look like they have very different purposes.  Should they perhaps have different names, e.g., `resizeForCopyAndPermute` and `resizeForUnpackAndCombine`?

Reply:
Ok, if that is preferred.  They both just generate the padding map for different inputs.  I can change their names.

Reply:
actually, their purposes are identical - compute the necessary padding and call resizeForIncomingDataImpl.  The different inputs are used to compute the padding in different ways.

Reply:
It's so tempting to merge `copyAndPermute` and `unpackAndCombine` in CrsGraph and CrsMatrix....

I would prefer a rename if you happen to be changing other things -- the names should make it more clear which one to call when.  Thanks!